EIPS/eip-1193.md

```diff
@@ -500,7 +500,7 @@ ethereum
       if (message.type === 'eth_subscription') {
         const { data } = message;
         if (data.subscription === subscriptionId) {
-          if (typeof data.result === 'string' && data.result) {
+          if (data.result && typeof data.result === 'object') {
```

Comment:
```suggestion
          if ('result' in data && typeof data.result === 'object') {
```
Consider being more explicit in what you are checking here, rather than using implicit conversions and fuzzy matching.